samples/lorawan/class_a/src/main.c

```diff
@@ -25,13 +26,27 @@ BUILD_ASSERT(DT_NODE_HAS_STATUS(DEFAULT_RADIO_NODE, okay),
 					  0x09, 0xCF, 0x4F, 0x3C }
 
 #define DELAY K_MSEC(10000)
+#define DL_MSG_LEN 256
 
 #define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
 #include <logging/log.h>
 LOG_MODULE_REGISTER(lorawan_class_a);
 
 char data[] = {'h', 'e', 'l', 'l', 'o', 'w', 'o', 'r', 'l', 'd'};
 
+void downlink_printer(uint8_t port, uint8_t length, uint8_t *payload)
+{
+	char hex_str[DL_MSG_LEN*2+1];
+	int ret;
+	ret = bin2hex(hex_str, DL_MSG_LEN*2+1, payload, length);
```

Comment:
You might want to use `LOG_HEXDUMP_INF` which does all the job of converting the data to hex format.

Reply:
I wanted to be able to format the log string with the port number the message arrived as and `LOG_HEXDUMP_INF` doesn't offer string formatting afaic?